File: Telenor.WebClient/src/app/components/products/viewmodels/product.ts
export class Product 
{
    Id: number;
    Name: string;
    Description: string;
    Price: number;
    PictureFileName: string;
    PictureUri: string;
    ProductFeatures:ProductFeatures [];
    ProductBrand: ProductBrand;
    ProductCategory: ProductCategory;
}
export class ProductBrand
{
    BrandName:string;
}
export class ProductCategory
{
    CategoryName:string;
}

export class ProductFeatures
{
    FeatureValue: string;
    ProductFeatureType: ProductFeatureType;
}
export class ProductFeatureType
{
    FeatureName:string;
}

File: Telenor.DbEntities/Product.cs
using System;
using System.Collections.Generic;

namespace Telenor.DbEntities
{
    public class Product
    {
        public int Id { get; set; }

        public string Name { get; set; }

        public string Description { get; set; }

        public decimal Price { get; set; }

        public string PictureFileName { get; set; }

        public int CategoryId { get; set; }

        public ProductCategory ProductCategory { get; set; }

        public int BrandId { get; set; }

        public ProductBrand ProductBrand { get; set; }

        public IEnumerable<ProductFeature> ProductFeatures { get; set; }
    }
}
